app/src/main/java/com/github/godspeed010/weblib/fragments/WebViewFragment.kt

```diff
@@ -213,19 +316,28 @@ class WebViewFragment : Fragment() {
         bottomNav.visibility = visibility
     }
 
-    override fun onStop() {
-        super.onStop()
-
-        overwriteSave()
-    }
-
     fun overwriteSave() {
         val folders = PreferencesUtils.loadFolders(activity)
 
         //update the url for the novel
-        folders[folderPosition].webNovels[novelPosition].url = lastVisitedUrl.toString()
+        folders[folderPosition].webNovels[novelPosition].url = lastVisitedUrl
+        folders[folderPosition].webNovels[novelPosition].progression = lastProgression
 
         //save the updated data
         PreferencesUtils.saveFolders(activity, folders)
+        Log.d(TAG, "Web Novel saved!")
+    }
+
+    private fun calculateProgression(wv: WebView): Float {
+        return (((wv.scrollY - wv.top - 200).toFloat() / resources.displayMetrics.density) / wv.contentHeight).coerceAtLeast(0f)
+    }
+
+    private fun calculateScrollYFromProgression(progression: Float, wv: WebView): Int {
+        return ((progression * wv.contentHeight) * resources.displayMetrics.density).toInt() + wv.top
+    }
+
+    companion object {
+        const val STATE_PROGRESSION = "novelProgression"
+        const val STATE_URL = "novelUrl"
```

Comment:
Don't see any reason to have a companion object. Also, there's no need to keep them public, so best to set them private.

 The Constants can be put above the Class declaration like so:

```
private const val STATE_PROGRESSION = "novelProgression"
private const val STATE_URL = "novelUrl"

class WebViewFragment : Fragment() {
//...
}
```

Reply:
Alright, I will move those to be global-level declarations instead of companion object variables. I think I was basing it off both the [examples ](https://developer.android.com/guide/components/activities/activity-lifecycle#save-simple,-lightweight-ui-state-using-onsaveinstancestate) and [Android's common Kotlin patterns guide](https://developer.android.com/kotlin/common-patterns#companion).